### DISCUSSION TITLE
Segfault in cairo_scaled_font_get_font_options

### DISCUSSION THREAD

**Discussion by rboati** [ghostty-org/ghostty#3258] [Feature Requests, Ideas] [answered]
Probably after a packages update (Debian unstable), I'm not able to start ghostty anymore. The crash is so bad that often it kills my session too and I have to logout. Updating ghostty or deleting configs and caches doesn't solve the problem. I already sent a couple of sentry reports.
The command journalctl -f shows this error:
```
dic 11 11:06:24 XXXXX  kernel: ghostty[35452]: segfault at 8 ip 00007fcc61ee35fc sp 00007ffc8ece0b80 error 4 in libcairo.so.2.11802.2[6e5fc,7fcc61e89000+ec000] likely on CPU 1 (core 1, socket 0)
```

As far as I know, the function `cairo_scaled_font_get_font_options(cairo_scaled_font_t *scaled_font, cairo_font_options_t *options) `receives as input `scaled_font` with value `NULL` (the parameter `options` instead seems to point to a valid instance) and goes into segfault as soon as it tries to extract the `status` field.

See also the small discord thread about it in [Ghostty started to crash at start some days ago](https://discord.com/channels/1005603569187160125/1316362922511040533)

**Comment by mitchellh** [comment]
Is this an upstream issue? We don't use Cairo at all. GTK does but we dynamically link to it so the GTK package would have to fix this. I'm happy to apply a workaround if we can find one. I read the entire Discord thread as well and couldn't glean any more info...

**Comment by nicholascapo** [comment]
I'm hitting this as well (also Debian testing/unstable), I submitted some Sentry reports, built with and without `-Doptimize=ReleaseFast`.
I also ran it under `gdb` with `debuginfod` running, so maybe those reports will have complete traces (?).
Is there more I can do to expand the report?

**Comment by weirdan** [comment]
Here's the backtrace, if it helps: https://gist.github.com/weirdan/705a9a73664b77145e1198740e25d906

**Reply by mitchellh** [reply]
This definitely looks like a GTK bug

**Comment by rcalixte** [comment, marked as answer]
I haven't been able to launch Ghostty for almost a month and then I added
```
gtk-titlebar = false
```
 to my config (it also works via the CLI) and Ghostty at least launches now. If I try to resize the window or open a second tab, Ghostty crashes though. The same with splits. So it's definitely a bug specific to the current Debian Testing (13/Trixie).<sup>1</sup> I'm afraid I don't have much more than that but hopefully that at least lets others use Ghostty for a little more.

I should also add that I had [this bug](https://github.com/ghostty-org/ghostty/issues/2470) which also appears to be isolated to the current Debian Testing (13/Trixie).

<sup>1</sup> This appears to be a [bug in Pango](https://github.com/ghostty-org/ghostty/discussions/3311).

**Reply by tbiehn** [reply]
Confirmed this issue is really pango - and resolved by the downgrade suggested here: https://github.com/ghostty-org/ghostty/discussions/3311#discussioncomment-11679029

**Reply by rboati** [reply]
I can confirm that downgrading **libpango** from 1.55.0+ds-2  to 1.54+ds-3 solved my problem!
As reported in bug #3311 the downgrading in Debian Sid worked smoothly and now Ghostty is working again.
Disabling the GTK Titlebar didn't have any effect on my system.

**Reply by rboati** [reply]
Now there is a better insight of this bug here [#3311](https://github.com/ghostty-org/ghostty/discussions/3311#discussioncomment-11687825) thanks to @Fil0sOFF

**Comment by rboati** [comment]
I think this discussion can be closed.
